_pytest/recwarn.py

```diff
@@ -216,7 +216,8 @@ def __exit__(self, *exc_info):
         # only check if we're not currently handling an exception
         if all(a is None for a in exc_info):
             if self.expected_warning is not None:
-                if not any(r.category in self.expected_warning for r in self):
+                if not any(issubclass(r.category, exp_warning) for
```

Comment:
lets use issubclass with the propper classinfo support here, no need for own loops, please check the docs